docs/design/service_traffic_policy.md

```diff
@@ -315,3 +315,134 @@ where packet leaves the node and goes back to the external entity that initiated
+
+```
+-A OUTPUT -j OVN-KUBE-ITP
+[1:60] -A OVN-KUBE-ITP -d 10.96.61.132/32 -p tcp -m tcp --dport 80 -j MARK --set-xmark 0x1745ec/0xffffffff
+```
+
+2. A routing policy (priority 30) is setup in the database to match on this mark and send it to custom routing table `number 7`:
+
+```
+root@ovn-worker:/# ip rule
+30:	from all fwmark 0x1745ec lookup 7
+```
+
+3. In routing table `7` we have a route that steers this traffic into management port `ovn-k8s-mp0` instead of sending it via default service route towards `breth0`:
+
+```
+root@ovn-worker:/# ip r show table 7
+10.96.0.0/16 via 10.244.0.1 dev ovn-k8s-mp0 
+```
+
+4. Packet enters ovn via `k8s-nodename` port and hits the load balancer on the switch where the packet is DNAT-ed into the local endpoint if any, else rejected.
+
+Here is a sample load balancer on `ovn-worker` node which has an endpoint:
+
+```
+_uuid               : d58c012a-7b1f-45ad-a817-a86845fde164
+external_ids        : {"k8s.ovn.org/kind"=Service, "k8s.ovn.org/owner"="surya/hello-world-2"}
+health_check        : []
+ip_port_mappings    : {}
+name                : "Service_surya/hello-world-2_TCP_node_switch_ovn-worker"
+options             : {event="false", reject="true", skip_snat="false"}
+protocol            : tcp
+selection_fields    : []
+vips                : {"10.96.61.132:80"="10.244.0.6:8080", "172.18.0.2:31358"="10.244.0.6:8080,10.244.1.3:8080", "172.18.0.9:80"="10.244.0.6:8080,10.244.1.3:8080"}
+```
+
+Note that only endpoints for `ClusterIP` are filtered, externalIPs/nodePorts are not.
+
+Here is a sample load balancer on `ovn-control-plane` node which does not have an endpoint:
+
+```
+_uuid               : a7865d9f-43d2-4cf9-a316-fc35e8c357a8
+external_ids        : {"k8s.ovn.org/kind"=Service, "k8s.ovn.org/owner"="surya/hello-world-2"}
+health_check        : []
+ip_port_mappings    : {}
+name                : "Service_surya/hello-world-2_TCP_node_switch_ovn-control-plane"
+options             : {event="false", reject="true", skip_snat="false"}
+protocol            : tcp
+selection_fields    : []
+vips                : {"10.96.61.132:80"="", "172.18.0.4:31358"="10.244.0.6:8080,10.244.1.3:8080", "172.18.0.9:80"="10.244.0.6:8080,10.244.1.3:8080"}
+```
+
+5. Packet is then delivered to backend pod.
+
+### **Host -> Service -> Host Networked Pod**
+
+When the backend is a host networked pod we shortcircuit OVN to counter reverse path filtering issues and use iptables rules on the host to DNAT directly to the correct host endpoint.
+
+```
+[1:60] -A OVN-KUBE-ITP -d 10.96.48.132/32 -p tcp -m tcp --dport 80 -j REDIRECT --to-ports 8080
+```
+
+NOTE: If a service with ITP=local has both host-networked pods and ovn pods as local endpoints, traffic will always be delivered to the host-networked pod. This is acceptable since traffic policy claims unfair load balancing as a side effect of the feature.
```

Comment:
why does ITP=local cause this to happen? Does ths happen for services without InternalTrafficPolicy?

Reply:
Nevermind, I understand now.

Reply:
because let's say you have a host-networked pod as backend and ovn pod as backend for the same service. The IPTable rule on the host with the REDIRECT will take precedence, flow won't hit the switch LB.

But I should probably clarify this is only for host-> svc case, not for pod->svc.